src/rgw/rgw_asio_client.cc

```diff
@@ -110,9 +110,8 @@ size_t ClientIO::read_data(char* buf, size_t max)
     boost::system::error_code ec;
     beast::http::read_some(socket, buffer, parser, ec);
     if (ec == boost::asio::error::connection_reset ||
-        ec == boost::asio::error::eof ||
-        ec == beast::http::error::partial_message ||
-        ec == beast::http::error::need_buffer) {
+        ec == beast::http::error::need_buffer ||
```

Comment:
So this means that your incoming HTTP message bodies cannot be larger than the buffer size?

Reply:
our request handlers will call this `read_data(char* buf, size_t max)` function repeatedly to read the body in pieces. as i understand it, by breaking on `need_buffer` here and returning successfully, we indicate to the caller that it needs to provide another buffer

Reply:
but the code also breaks on `connection_reset`, is the intention to treat `connection_reset` the same as `need_buffer`?

Reply:
agreed, `connection_reset` doesn't belong here. it was only working because the request handler's loop terminates on a zero-length read. after removing that check for `connection_reset`, i found that we have one s3test case, `test_atomic_dual_conditional_write_1mb`, that exercises that error path and logs `failed to read body: Connection reset by peer`

pr updated - thanks again!

Reply:
Nice!